lib/std/fs.zig

```diff
@@ -1587,3 +1587,57 @@ test "" {
     _ = @import("fs/file.zig");
     _ = @import("fs/get_app_data_dir.zig");
 }
+
+// Custom read-by-line iterator with some FP-style secret sauce
+// Used for std.terminfo.Terminfo
+pub const ReadLineIterator = struct {
+    const Self = @This();
+    const FilterCallback = fn([]u8)bool;
+    const MapCallback = fn([]u8)[]const u8;
+
+    in_stream: File.InStream,
+    buf_in_stream: std.io.BufferedInStream(File.ReadError),
+    filter_fn: FilterCallback,
+    map_fn: MapCallback,
+
+    pub fn init(file_path: []const u8) !Self {
+        const file = try std.fs.cwd().openFile(file_path, .{});
+        var in_stream = file.inStream();
+
+        return Self {
+            .in_stream = in_stream,
+            .buf_in_stream = std.io.BufferedInStream(File.ReadError).init(&in_stream.stream),
+            .filter_fn = struct { fn filter(line: []u8) bool { return true; } }.filter,
+            .map_fn = struct { fn map(line: []u8) []u8 { return line; } }.map,
+        };
```

Comment:
you should be able to return a anon struct literal here and `filter_fn` and `map_fn` can probably be set as default struct values since the functions are comptime known.